### ISSUE TITLE
App: Hide the buttons when not in use

### ISSUE THREAD

**Issue by zstadler** [IsraelHikingMap/Site#1180]
## What I expect to happen
When the user stops interacting with the app, hide the buttons in order to show a larger portion of the map.

## What really happened
The buttons are taking precious screen real-estate and hide portions of the map. This is especially important for newer models where the screen ratio is even narrower.

The request was raised by several posts on Jeepolog:
- [הלחצנים גוזלים מקום יקר על המסך. אפשר לקחת דוגמא מאורוקס ולהעלים אותם כשהם לא בשימוש](http://www.jeepolog.com/forums/showthread.php/115410-%D7%A9%D7%97%D7%A8%D7%A8%D7%A0%D7%95-%D7%90%D7%AA-%D7%9E%D7%A4%D7%95%D7%AA-%D7%94%D7%98%D7%99%D7%95%D7%9C%D7%99%D7%9D-%D7%94%D7%9B%D7%99-%D7%98%D7%95%D7%91%D7%95%D7%AA-%D7%A9%D7%9C-%D7%99%D7%A9%D7%A8%D7%90%D7%9C-%D7%9E%D7%A2%D7%95%D7%9C%D7%9D?p=881666&viewfull=1#post881666)
- [הלחצנים גם לדעתי רובם מיותרים- לפחות בזמן שלא נוגעים במסך](http://www.jeepolog.com/forums/showthread.php/115410-%D7%A9%D7%97%D7%A8%D7%A8%D7%A0%D7%95-%D7%90%D7%AA-%D7%9E%D7%A4%D7%95%D7%AA-%D7%94%D7%98%D7%99%D7%95%D7%9C%D7%99%D7%9D-%D7%94%D7%9B%D7%99-%D7%98%D7%95%D7%91%D7%95%D7%AA-%D7%A9%D7%9C-%D7%99%D7%A9%D7%A8%D7%90%D7%9C-%D7%9E%D7%A2%D7%95%D7%9C%D7%9D?p=881667&viewfull=1#post881667)
- [++1
בזמן ניווט, רוב זמן השימוש, כל הכפתורים מיותרים. תלמדו מ 
Oruxmaps 
(ועמוד ענן).](http://www.jeepolog.com/forums/showthread.php/115410-%D7%A9%D7%97%D7%A8%D7%A8%D7%A0%D7%95-%D7%90%D7%AA-%D7%9E%D7%A4%D7%95%D7%AA-%D7%94%D7%98%D7%99%D7%95%D7%9C%D7%99%D7%9D-%D7%94%D7%9B%D7%99-%D7%98%D7%95%D7%91%D7%95%D7%AA-%D7%A9%D7%9C-%D7%99%D7%A9%D7%A8%D7%90%D7%9C-%D7%9E%D7%A2%D7%95%D7%9C%D7%9D?p=881700&viewfull=1#post881700)

## Thing I think the developers should know, images, links etc.
There are several options for the Ux of showing the buttons after they were hidden
- An icon, possibly shaped as a triangle or V, that will cause the buttons to show. Similar to OruxMaps.
- A touch of the screen would show the hidden buttons
- A touch at the buttons area of the screen would show the hidden buttons
and probably more

**Comment by HarelM** [comment]
I've read the posts and also other post those people wrote.
These are not the typical users - they are used to Orux and its advance functionality.
Hiding buttons is just bad UX.
The only solution I can think of is by using a menu style button, which I don't think improves the UX as every operation after an idle time consts two clicks which is annoying IMHO, my orux buttons where always visible when I used Orux.
The map is not the only thing relevant in the app, although this is the only thing that interest these guys.
Neither Waze nor off-road is hiding buttons, nor most of the applications why not learn from them?

**Comment by zstadler** [comment]
I beg to differ. 

A. These are constructive proposals from people who care and want to contribute to the improvement of the Israel Hiking App.

B. The approaches  for increasing the screen size allocated to the map differ from one application to another, but every app I've checked does its best to maximize it:

- Waze
- Off-Road
- Google Maps
- OruxMaps
- Lucus Map
- Amud Anan

I think we need to do it too.

**Comment by HarelM** [comment]
I dont believe hiding buttons is the right solution.
I can't think of a good one without moving a lot of buttons to some menus, which might solve this problem but will create other countless problems.
Feel free to suggest a UX for this solution without hiding buttons and I'll be able to address it. Otherwise we're just waiting or breath.

**Comment by valleyofdawn** [comment]
I think we should find a middle ground.
We can probably agree that:
1) Map real-estate is precious
2) Too many buttons are confusing
3) Crowded and small buttons make it hard to see and operate the app in the field

Using the 5+/-2 screen element rule I think we should keep only the buttons you are likely to use while navigating on top.
The ones used rarely, or used only when you plan a route at the comfort of your home, should go under a hamburger button.
Something along this line:
![image](https://user-images.githubusercontent.com/689201/76746715-9e727f00-6780-11ea-9f96-ecd2497b2742.png)

**Comment by HarelM** [comment]
There's no chance we could agree on the important buttons as every one of us needs different buttons.
For example:
GPS position is redundant to me - the app should just show the position at all times.
Keep north up is redundant to me - never used it.
Save and Open file is redundant to me - I use shares.
Zoom-in and out is redundant to me - I use gesture.
Info can be inside the user profile or somewhere hidden.
Distance, graph etc is redundant to me in most cases - I rarely use it.
Save as is redundant to me - I never use it.
Layer can probably by inside the user profile/hamburger button.
Language too.
etc, etc.
Bottom line, what you would like and the users that wrote the post is to re-implement oruxmaps, I would like to avoid it.
This discussion is pointless as I don't see a way we could agree, everyone of us needs a different flavor as we are using the app in a different way and I'm not planning on implementing a button selector like there is in Orux.
If you would like to do a survey in facebook to see the most used features for users and keep only the top 5 to 7 in the main screen I'm fine with that assuming we will honor the results, otherwise it's a wait of time.

**Comment by valleyofdawn** [comment]
Can we get stats on what is being pressed right now?
We can agree on a number of top-screen buttons and then each suggest an ordered  list of favorites. This will let us build a compromise list we can agree in advance to accept.
I for one would not like to recreate Orux. I do think we can learn from it as well as from other popular apps.

**Comment by HarelM** [comment]
Feel free to research our google analitics account. I couldn't find a good way to split the button presses data by platform (web/android/ios), I also didn't spend a lot of time trying.
If you find a way let me know.

**Comment by HarelM** [comment]
I also do not agree on the basic assumption the map real estate is that precious to me. Using a bottom bar, top bar and side buttons is ok by me keeping an area of 60-70% for the map is sufficient to me...

**Comment by HarelM** [comment]
I have removed all of my comments.

**Comment by valleyofdawn** [comment]
I tried to get statistics on what is being pressed but
1) I couldn't separate App use from site use
2) Many of the event names were obscure
I think we agree more than you think on what should go under the "..." button and what should stay on top. Even if we don't agree fully we could put it to a vote.
Let's first decide on how many top-level buttons we should have.
I suggest 7. True - it's a little too many based on most app-design guidelines, but it's an improvement over the 20 (!) we have now.

**Comment by valleyofdawn** [comment]
@zstadler @HarelM 
Any thoughts on the ideal number of buttons?

**Comment by HarelM** [comment]
Let's take this discussion off-line please.
